frontend/tests/controllers/ProblemCreateTest.php

```diff
@@ -1160,4 +1160,154 @@ public function testCreateProblemWithAllowUserAddTagsValues(
+            ],
+            'guest2' => [
+                'problems' => [],
+            ],
+        ];
+
+        $guestUsers = ['guest1', 'guest2'];
+
+        $expectedProblemsBeforeAdmins = [
+            'guest1' => ['owner1_public', 'owner2_public'],
+            'guest2' => ['owner1_public', 'owner2_public'],
+        ];
+
+        $expectedProblemsAfterAdmins = [
+            'guest1' => ['owner1_private', 'owner1_public', 'owner2_public'],
+            'guest2' => ['owner1_public', 'owner2_private', 'owner2_public'],
+        ];
+
+        // Create all users
+        $users = [];
+        foreach (array_keys($testScenario) as $userName) {
+            ['identity' => $users[$userName]] = \OmegaUp\Test\Factories\User::createUser();
+        }
+
+        // Create all problems
+        $problems = [];
+        foreach ($testScenario as $ownerName => $ownerData) {
+            foreach ($ownerData['problems'] as $problemAlias => $problemData) {
+                $problems[$problemAlias] = \OmegaUp\Test\Factories\Problem::createProblem(
+                    new \OmegaUp\Test\Factories\ProblemParams([
+                        'author' => $users[$ownerName],
+                        'visibility' => $problemData['visibility'],
+                        'alias' => $problemAlias,
+                    ])
+                );
+            }
+        }
+
+        // Verify that each guest only sees public problems (before adding permissions)
+        foreach ($guestUsers as $guestName) {
+            $guestLogin = self::login($users[$guestName]);
+            $response = \OmegaUp\Controllers\Problem::apiList(
+                new \OmegaUp\Request(['auth_token' => $guestLogin->auth_token])
+            );
+            $list = array_map(
+                fn($problem) => $problem['alias'],
+                $response['results']
+            );
+
+            foreach ($testScenario as $ownerData) {
+                foreach ($ownerData['problems'] as $problemAlias => $problemData) {
+                    $shouldSee = in_array(
+                        $problemAlias,
+                        $expectedProblemsBeforeAdmins[$guestName]
+                    );
+
+                    if ($shouldSee) {
+                        self::assertArrayContainsWithPredicate(
+                            $list,
+                            fn(string $a) => $a == $problemAlias,
+                            "{$guestName} should see problem {$problemAlias}"
+                        );
+                    } else {
+                        self::assertArrayNotContainsWithPredicate(
+                            $list,
+                            fn(string $a) => $a == $problemAlias,
+                            "{$guestName} should not see problem {$problemAlias}"
+                        );
+                    }
+                }
+            }
+        }
+
+        // Add admins to problems according to configuration
+        foreach ($testScenario as $ownerData) {
```

Comment:
Este nos falataba ajustar:
```suggestion
        foreach ($testScenario['users'] as $ownerData) {
```